referenceqvm/unitary_generator.py

```diff
@@ -355,6 +356,45 @@ def tensor_gates(gate_set, defgate_set, pyquil_gate, num_qubits):
     return gate
 
 
+def tensor_up(pauli_terms, num_qubits):
+    """
+    Takes a PauliSum object along with a total number of
+    qubits and returns a matrix corresponding the tensor representation of the
+    object.
+
+    Useful for generating the full Hamiltonian after a particular fermion to
+    pauli transformation.
+
+    :param pauli_terms: (PauliSum) object of PauliTerm
+    :param num_qubits: (int) number of qubits in the system
+    :returns: (numpy array) representation of the paui_terms operator
+    """
+    if not isinstance(pauli_terms, PauliSum):
+        raise TypeError("can only tensor PauliSum")
+
```

Comment:
You might wanna check for sanity if the operator gets too big. If you kron up 100 terms then the final matrix will be gargantuan and might kill you.
Otherwise you can always check for sparsity of the gate_matrices.

Reply:
We could do that.  I believe in the `qam.py` there was a hard coded limit of 51 qubits (way beyond what could be represented with a numpy array on a typical machine).  I advocated against  putting a limit because the formal structure of the QAM doesn't have a limit.

Reply:
I can see that argument. Maybe we can find a middle ground and generate a warning (https://docs.python.org/3/library/warnings.html#module-warnings) beyond some semi-reasonable limit. This will not interrupt the code execution but warn the user that code with large number of qubits might have a performance issue